lib/rust/metamodel/src/java/syntax.rs

```diff
@@ -211,10 +226,22 @@ impl fmt::Display for Type {
 
 impl fmt::Display for Method {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let Method { name, arguments, return_, static_, final_, body, override_, throws } = &self;
+        let Method {
+            name,
+            arguments,
+            return_,
+            static_,
+            final_,
+            body,
+            override_,
+            throws,
+            visibility,
+        } = &self;
         let mut tokens = vec![];
         override_.then(|| tokens.push("@Override".to_string()));
-        tokens.push("public".to_string());
+        if let Some(visibility) = visibility {
```

Comment:
I have to admit I have no idea how a statement like this can even type correctly!

```
if let Some(visibility) = visibility
```

Am I equal to something that is clearly not equal to me? However otherwise the check is right - only if things are supposed to be visible, we add the "public" modifier.

Reply:
`if let` introduces a conditional destructuring binding, for a refutable pattern. The `visibility` on the right-hand-side of the expression is an `Option<Visibility>`, so it has two possibilities: `Some(value)`, or `None`. The block following the `if let` will be entered if `visibility` is `Some(anything)`; otherwise, the pattern doesn't match and the block doesn't run. The identifier within the `Some(_)` pattern determines the binding of the newly-unwrapped value, within the block. When destructuring an `Option` like this, it is common practice to shadow the variable being matched with the new binding, so within the block, `visibility` will refer to the `Visibility` value contained by the `Option<Visibility>`. I think the Java equivalent would be something like:
```java
if (visibility != null) {
    tokens.add(visibility.toString())
}
```
Except in Rust the `Option` is enforced, so it wouldn't be possible use the "contents" of the option without the `if let` (or some other means of predicating on the `Option` status).

Reply:
I was always wondering how to get value from an `Option`!? All those `.map` & `.filter` drove me crazy. I have to remember `if let Some(o) = o` as that seems like the straighforward way for a Java guy.

Reply:
I was always wondering how to get value from an `Option`!? All those `.map` & `.filter` drove me crazy. I have to remember `if let Some(o) = o` as that seems like the straighforward way for a Java guy.